whois-commons/src/main/java/net/ripe/db/whois/common/apiKey/ApiKeyUtils.java

```diff
@@ -24,38 +22,20 @@ public class ApiKeyUtils {
 
     public static final String APIKEY_KEY_ID_QUERY_PARAM = "keyId";
 
-    public static boolean validateScope(final OAuthSession oAuthSession, final List<RpslObject> maintainers) {
-        if(StringUtils.isEmpty(oAuthSession.getScope())) {
-            return true;
-        }
-
-        final Optional<String> whoisScope = getWhoisScope(oAuthSession);
-        if(whoisScope.isEmpty()) {
-            return true;
-        }
-
-        final OAuthSession.ScopeFormatter scopeFormatter = new OAuthSession.ScopeFormatter(whoisScope.get());
-
-        if(StringUtils.isEmpty(scopeFormatter.getScopeKey()) || StringUtils.isEmpty(scopeFormatter.getScopeType()) || StringUtils.isEmpty(scopeFormatter.getAppName())) {
-            return true;
-        }
+    public static boolean validateMntnrScope(final OAuthSession oAuthSession, final List<RpslObject> maintainers) {
+        final List<String> mntnrScopes =  oAuthSession.getScopes().stream()
```

Comment:
I think that in a previous step we are sending empty scopes in case no scope starts by "whois"
so this will be empty at this point so this method will return true?

Reply:
Yes, mntnr scope is not mandatory